### PR TITLE
Use DimPlugSets.

### PR DESCRIPTION
**Pull request by ryan-rushton**
This cleans up a few places I found using the PlugSet def or the plug sets with unlocked plugs per profile. We can probably do more around the profile stuff.

### CODE REVIEW

File: src/app/inventory/store/energy.ts [hunk 1]
```diff
@@ -30,22 +29,18 @@ export function energyUpgrade(
...
-  const plugSet = defs.PlugSet.get(tierSocket.socketDefinition.reusablePlugSetHash);
-
   const energyMods: DestinyInventoryItemDefinition[] = [];
-  for (const { plugItemHash } of plugSet.reusablePlugItems) {
-    const plugItem = defs.InventoryItem.get(plugItemHash);
-
-    const capacity = plugItem.plug?.energyCapacity;
-    if (!plugItem.plug || !capacity) {
+  for (const dimPlug of tierSocket.plugSet.plugs) {
+    const capacity = dimPlug.plugDef.plug?.energyCapacity;
```
Comment (nev-r) [inline src/app/inventory/store/energy.ts:38]: i think if you check types this optional chain is no longer necessary

File: src/app/inventory/store/energy.ts [hunk 2]
```diff
@@ -30,22 +29,18 @@ export function energyUpgrade(
...
+    if (!dimPlug.plugDef.plug || !capacity) {
```
Comment (nev-r) [inline src/app/inventory/store/energy.ts:39]: and this check could be scaled down

File: src/app/loadout-builder/item-filter.ts [hunk 3]
```diff
@@ -117,40 +117,30 @@ function matchedLockedModEnergy(
...
-        // but the API considers it to be disabled.
-        socket.plugged &&
-        defs.PlugSet.get(socket.socketDefinition.reusablePlugSetHash).reusablePlugItems.map(
-          (plugItem) => plugItem.plugItemHash
-        )
-    )
-  ).sort(compareBy((plugHashes) => plugHashes.length));
+  const socketsOrderedWithArtificerFirst = sockets
+    // If a socket is not plugged (even with an empty socket) we consider it disabled
+    // This needs to be checked as the 30th anniversary armour has has the Artificer socket
```
Comment (nev-r) [inline src/app/loadout-builder/item-filter.ts:129]:
```suggestion
    // This needs to be checked as the 30th anniversary armour has the Artifice socket
```

File: src/app/loadout-builder/item-filter.ts [hunk 4]
```diff
@@ -117,40 +117,30 @@ function matchedLockedModEnergy(
...
-    const plugSetIndex = plugSets.findIndex((set) => set.includes(mod.hash));
-    if (plugSetIndex === -1) {
+    const socketIndex = socketsOrderedWithArtificerFirst.findIndex((socket) =>
+      socket.plugSet?.plugs.some((plug) => plug.plugDef.hash === mod.hash)
+    );
+    if (socketIndex === -1) {
       return false;
     }
-    plugSets.splice(plugSetIndex, 1);
+    socketsOrderedWithArtificerFirst.splice(socketIndex, 1);
```
Comment (nev-r) [inline src/app/loadout-builder/item-filter.ts:143]:
```suggestion
    socketsOrderedWithArtificeFirst.splice(socketIndex, 1);
```

File: src/app/loadout-builder/item-filter.ts [hunk 5]
```diff
@@ -117,40 +117,30 @@ function matchedLockedModEnergy(
...
+    // but the API considers it to be disabled.
+    .filter((socket) => socket.plugSet && socket.plugged)
+    // Artificer sockets only plug a subset of the bucket specific mods so we sort by the size
+    // of the plugItems in the plugset so we use that first if possible.
+    .sort(compareBy((socket) => socket.plugSet?.plugs.length));
 
   for (const mod of lockedMods) {
-    const plugSetIndex = plugSets.findIndex((set) => set.includes(mod.hash));
-    if (plugSetIndex === -1) {
+    const socketIndex = socketsOrderedWithArtificerFirst.findIndex((socket) =>
```
Comment (nev-r) [inline src/app/loadout-builder/item-filter.ts:137]:
```suggestion
    const socketIndex = socketsOrderedWithArtificeFirst.findIndex((socket) =>
```

**Review by nev-r — COMMENTED** [review]
`filterDimPlugsAvailableToCharacerOrProfile` needs a global spellcheck
or maybe just a name like `filterDimPlugsForCharacter`

File: src/app/records/plugset-helpers.ts [hunk 6]
```diff
@@ -1,22 +1,6 @@
...
-  profileResponse: DestinyProfileResponse,
-  plugSetHash: number
-) {
-  return (profileResponse.profilePlugSets.data?.plugs[plugSetHash] || []).concat(
-    Object.values(profileResponse.characterPlugSets.data || {})
-      .filter((d) => d.plugs?.[plugSetHash])
-      .flatMap((d) => d.plugs[plugSetHash])
-  );
-}
-
```
Comment (ryan-rushton) [inline src/app/records/plugset-helpers.ts:19]: I deleted this as its no longer used but I will need to create another check for the subclass stuff to ensure aspects/fragments are unlocked before I can merge this.

File: src/app/inventory/store/energy.ts (hunk 2 above)
Comment (ryan-rushton) [reply src/app/inventory/store/energy.ts]: Done

File: src/app/inventory/store/energy.ts (hunk 1 above)
Comment (ryan-rushton) [reply src/app/inventory/store/energy.ts]: Done

**Review by nev-r — APPROVED** [review]

**Pull request merged**